app/validators/form_email_validator.rb

```diff
@@ -15,6 +16,15 @@ module FormEmailValidator
 
   private
 
+  def validate_domain
+    return unless email.present?
+    domain = email&.split('@')&.last
```

Comment:
I would use a parser instead of a split, but otherwise same same
```suggestion
    domain = Mail::Address.new(email).domain
```

Reply:
Ah, thanks, today I learned.

Auto-converting to punycode occurred to me as well, but was focused on the unhandled exception for now.

Reply:
auto-converting could make for a good follow-up